earth_enterprise/tutorial/SConstruct

```diff
@@ -34,30 +34,62 @@ if (not installdir) and ('install' in COMMAND_LINE_TARGETS):
 tutorials_dir = installdir + '/tutorial'
 root_dir = tutorials_dir + '/opt/google/share/tutorials/fusion/'
 
-def makeHardLinks(instdir, target_dir, excluded_list):
-    # get contents of target_dir and make hard links to files in instdir
+# Get the current directory, since SCons will move into the directory
+#   it's using, and FusionTutorial is expected below the SConscript file.
+tutorial_src_dir = os.sep.join((os.getcwd().split(os.sep) +
+                    ['FusionTutorial']))
+
+def makeHardLinks(install_dir, target_dir, excluded_list):
+    # get contents of target_dir and make hard links to files in install_dir
     # recurse down the directories
 
     contents = os.listdir(target_dir)
-    target_dir = target_dir + "/"
+    target_dir = target_dir + "/"	
     # Ensure the install dir exists at each step
-    commands.getstatusoutput("mkdir -p " + instdir)
+    commands.getstatusoutput("mkdir -p " + install_dir)
     for file in contents:
         if file in excluded_list:
             continue
         target_file = target_dir + file
         if os.path.isdir(target_file):
-            subdir = instdir + "/" + file
+            subdir = install_dir + "/" + file
             makeHardLinks(subdir, target_file,
                                              excluded_list)
         else:
             commands.getstatusoutput("ln \"" + target_file + "\" " +
-                                     instdir)
+                                     install_dir)
+
+
+def makeSoftLinks(dest_dir, src_dir, excluded_list):
+    """Make soft links of all files in src_dir and place
+    them in dest_dir. Don't link to files in excluded_list.
+    Recurse through all subdirectories of src_dir.
+    """
+    # Strip trailing slashes because they cause trouble later
+    dest_dir = dest_dir.rstrip(os.sep)
+    src_dir = src_dir.rstrip(os.sep)
+    src_contents = os.listdir(src_dir)
+    # Ensure the destination directory exists at each step
+    commands.getstatusoutput("mkdir -p " + dest_dir)
+    for file in src_contents:
```

Comment:
Minor: `os.walk()` would work great for this purpose. On the other hand, this pattern matches the function above it. I'll leave it up to you whether you think it's worth changing or not.

Reply:
I'd prefer to match the original pattern. There are a few other optimizations that could be done here (see also my TODO about `commands` being deprecated), which could make a good issue filed as an 'improvement' vs. an 'issue'.